File: ROID-Forum-Web-Client/Convert Client Code/GMGWebsite/web/Website/Sections/GraphicsSection/CommentInfo.dart
library GraphicsCommentInfo;
import 'ThreadViews/CommentView.dart';
import 'ThreadController.dart';
import 'ThreadInfo.dart';

class CommentInfo {
  int _threadID;
  int _commentID;
  String _comment;
  String _owner;
  String _avatarURL = "";
  CommentView commentView;
  ThreadInfo thread;
  ThreadController threadController;
  CommentInfo(ThreadInfo t, ThreadController tc) {
    thread = t;
    threadController = tc;
    commentView = new CommentView(this, thread, threadController);
  }
  void setThreadID(int id) {
    _threadID = id;
  }int getThreadID() {return _threadID;}
  void setCommentID(int id) {
    _commentID = id;
  }int getCommentID() {return _commentID;}
  void setComment(String comment) {
    _comment = comment; commentView.updateDescription();
  }String getComment() {return _comment;}
  void setOwner(String owner) {
    _owner = owner; commentView.updateOwner();
  }String getPosterName() {return _owner;}
  void setAvatarURL(String avatarURL) {
        _avatarURL = avatarURL;
        commentView.updateAvatarURL();
      }
      String getAvatarURL() {
        return _avatarURL;
      }
}

File: ROID-Forum-Web-Client/src/Interface/CodingSection/CommentInfo.ts
import {CommentView} from "./ThreadViews/CommentView";
import {ThreadInfo} from "./ThreadInfo";
import {ThreadController} from "./ThreadController";

export class CommentInfo {
    _threadID: number;
    _commentID: number;
    _comment: string;
    _owner: string;
    _avatarURL = "";
    commentView: CommentView;
    thread: ThreadInfo;
    threadController: ThreadController;

    constructor(t: ThreadInfo, tc: ThreadController) {
        this.thread = t;
        this.threadController = tc;
        this.commentView = new CommentView(this, this.thread, this.threadController);
    }

    setThreadID(id: number) {
        this._threadID = id;
    }

    getThreadID(): number {
        return this._threadID;
    }

    setCommentID(id: number) {
        this._commentID = id;
    }

    getCommentID(): number {
        return this._commentID;
    }

    setComment(comment: string) {
        this._comment = comment;
        this.commentView.updateDescription();
    }

    getComment(): string {
        return this._comment;
    }

    setOwner(owner: string) {
        this._owner = owner;
        this.commentView.updateOwner();
    }

    getPosterName(): string {
        return this._owner;
    }

    setAvatarURL(avatarURL: string) {
        this._avatarURL = avatarURL;
        this.commentView.updateAvatarURL();
    }

    getAvatarURL(): string {
        return this._avatarURL;
    }
}